getnacelle/nacelle-js
# chore(shopify-cart): update deps for shopify-cart

## WHY are these changes introduced?

Addresses [ENG-8129](https://nacelle.atlassian.net/browse/ENG-8129) - moves us to the latest graphql-codegen updates 



## WHAT is this pull request doing?

- Updates graphql-codegen dependencies
- Updates other dependencies to latest versions
  - The only dependencies still called out by `npm outdated` are jest dependencies which had breaking changes. Which made them a bigger lift than covered by this PR. 


## How to Test

1. Checkout this branch `mdarrik/ENG-8129-update-codgen-deps`
2. Run `npm run clean && npm run bootstrap` from the root of the repo (optionally, run `CI=true npm run bootstrap` if you want to run CI resolution)
3. Navigate to `packages/shopify-cart`
4. Run `npm run codegen` should complete successfully and result in no file changes
5. Run `npm run build` should complete successfully
6. Run `npm run test:coverage` all tests should pass
